src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2688,7 +2688,7 @@ def list_dns_zones(resource_group_name=None):
 def create_dns_record_set(resource_group_name, zone_name, record_set_name, record_set_type,
                           metadata=None, if_match=None, if_none_match=None, ttl=3600):
     ncf = get_mgmt_service_client(DnsManagementClient).record_sets
-    record_set = RecordSet(name=record_set_name, type=record_set_type, ttl=ttl, metadata=metadata)
+    record_set = RecordSet(ttl=ttl, metadata=metadata)
```

Comment:
just to confirm, the `name` and `type` are indeed useless across all api-versions?

Reply:
They were made read only.  You still have to specify them, but you can't specify them in the RecordSet constructor anymore.  It was a rather annoying breaking change on the CLI side.  If you retrieve a record set, you can access name and type from the object.